### PR TITLE
Expose vendor cni dir prefix

### PR DESCRIPTION
**Pull request by abhinavdahiya**
Adds #46610 
**What this PR does / why we need it**:
- It adds `cni-vendor-dir-prefix` flag to kubelet to expose `cniVendorDirPrefix` for `VendorCNIDirTemplate` in https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/network/cni/cni.go#L39

- also makes changes to kubelet/dockershim to use the `cni-vendor-dir-prefix`

### CODE REVIEW

File: cmd/kubelet/app/options/options.go [hunk 1]
```diff
@@ -235,6 +235,7 @@ func (c *kubeletConfiguration) addFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&c.NetworkPluginDir, "network-plugin-dir", c.NetworkPluginDir, "<Warning: Alpha feature> The full path of the directory in which to search for network plugins or CNI config")
 	fs.StringVar(&c.CNIConfDir, "cni-conf-dir", c.CNIConfDir, "<Warning: Alpha feature> The full path of the directory in which to search for CNI config files. Default: /etc/cni/net.d")
 	fs.StringVar(&c.CNIBinDir, "cni-bin-dir", c.CNIBinDir, "<Warning: Alpha feature> The full path of the directory in which to search for CNI plugin binaries. Default: /opt/cni/bin")
+	fs.StringVar(&c.CNIVendorDirPrefix, "cni-vendor-dir-prefix", c.CNIVendorDirPrefix, "<Warning: Alpha feature> The full path of parent directory that houses all vendor CNI plugin binaries according to VendorCNIDirTemplate(\"<this-prefix>/opt/<cni-plugin-type>/bin\").  Default: \"\"")
```
Comment (liggitt) [inline cmd/kubelet/app/options/options.go]: Appending "/opt" is odd. Why not the dir searched for plugins, defaulting to "/opt"?

Comment (abhinavdahiya) [inline cmd/kubelet/app/options/options.go]: I am not sure what the original motivation to use `%s/opt/%s/bin` as VendorCNIDirTemplate (https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/network/cni/cni.go#L39)
I think conventionally when a host directory is mounted into a container a similar template is followed. (ex. `/opt/calico/bin` from the host is mounted into install-calico container as `/host/opt/calico/bin`)

**Comment by luxas** [comment]
Is this targeted for v1.7?

**Comment by freehan** [comment]
@k8s-bot ok to test

**Comment by dims** [comment]
@abhinavdahiya  Can you rebase please?

**Comment by luxas** [comment]
cc @kubernetes/sig-network-misc Do we need this anymore?

**Comment by caseydavenport** [comment]
I'm not sure I understand this PR. We already have `--cni-bin-dir` to allow specification of a CNI directory. Does this PR allow us to do something that we can't do today?

**Comment by feiskyer** [comment]
Seems `--cni-bin-dir` is enough for configuring external cni plugins. 

@abhinavdahiya Would you mind close the PR?

**Pull request closed without merging**